web/src/routes/PoisPage.tsx

```diff
@@ -78,27 +78,22 @@ const PoisPage = ({ cityCode, languageCode, cityModel, pathname, languages }: Ci
     }
   }, [])
 
-  const setCurrentFeature = (feature: PoiFeature | null) => {
-    const newQueryParams = queryParams
-    // Always deselect the old feature to allow flying to the new feature on the map. Otherwise flyTo somehow does not work.
-    newQueryParams.delete(locationName)
-    setQueryParams(newQueryParams)
-
+  const selectFeature = (feature: PoiFeature | null) => {
     if (feature) {
-      // Make sure old feature is actually deselected before selecting the new feature to fix flyTo.
-      setTimeout(() => {
-        newQueryParams.set(locationName, feature.properties.urlSlug)
-        setQueryParams(newQueryParams)
-      }, 1)
+      queryParams.set(locationName, feature.properties.urlSlug)
+    } else {
+      queryParams.delete(locationName)
     }
+    setQueryParams(queryParams)
   }
-
   useEffect(() => {
-    if (currentFeature) {
-      const coordinates = currentFeature.geometry.coordinates
-      flyTo(coordinates)
+    const currFeature =
+      featureLocations?.find((feature: PoiFeature) => feature.properties.urlSlug === selectedFeatureSlug) ?? null
+    if (currFeature) {
+      flyTo(currFeature.geometry.coordinates)
     }
-  }, [currentFeature, flyTo])
+    setCurrentFeature(currFeature)
```

Comment:
Why not just all this in `selectFeature`? You have the feature already there.

Reply:
because this `queryParams` hook has to be set with a timeout then to not intercept the setCurrentFeature behavior. So i separated the queryParams update from the currentFeature update.